higher_health/templates/healthcheck_questionnaire.html

```diff
@@ -292,33 +302,71 @@ <h3 class="heading-subtitle heading-subtitle--edu-label">Confirmation</h3>
     showHiddenInputs(selected_province.value, uniClone, uni, universities);
     showHiddenInputs(uni.options[uni.selectedIndex].value, campClone, camp, campuses);
     clearCampus();
+    showOther(selected_province.value, universities, uni_other);
+    showOther(uni.options[uni.selectedIndex].value, campuses, camp_other);
+
     prov.addEventListener('change', function() {
       showHiddenInputs(this.value, uniClone, uni, universities);
       showHiddenInputs(uni.value, campClone, camp, campuses);
       clearCampus();
     });
     uni.addEventListener('change', function() {
       showHiddenInputs(this.value, campClone, camp, campuses);
+      camp.parentElement.classList.remove('hidden');
+      showOther(this.value, universities, uni_other);
+    });
+    camp.addEventListener('change', function() {
+      showOther(this.value, campuses, camp_other);
     });
 
     function showHiddenInputs(elem, clonedElems, selectInputs, csvInputElem) {
       selectInputs.options.length = 0;
+      let last_option = null;
 
       for(let i = 0; i < clonedElems.options.length; i++) {
         let uniChild = clonedElems.options[i];
-        if(csvInputElem[uniChild.value] == elem || csvInputElem[uniChild.value] == undefined) {
-          selectInputs.appendChild(uniChild.cloneNode(true));
+
+        if(
+          csvInputElem[uniChild.value] == elem ||
+          csvInputElem[uniChild.value] == undefined ||
+          (uniChild.value == other_university && csvInputElem[uniChild.value] == '') ||
+          csvInputElem[uniChild.value] == other_university
+          ) {
+            if((uniChild.value == other_university && csvInputElem[uniChild.value] == '') ||
+             csvInputElem[uniChild.value] == other_university
+            ){
+              // if its the other option save it for last
+              last_option = uniChild;
+            }
+            else {
+              selectInputs.appendChild(uniChild.cloneNode(true));
+            }
         }
       }
+      // make sure we always add other as last opt
+      if (last_option !== null){
+        selectInputs.appendChild(last_option.cloneNode(true));
+      }
     };
+
+    function showOther(val, list, other){
+      if(val === other_university || String(list[val]) === other_university) {
```

Comment:
What is the `String(list[val])` doing here? I don't think it's very clear what we're trying to do here.